backend/lib/deal-observer.js

```diff
@@ -89,34 +74,37 @@ export async function storeActiveDeals (activeDeals, pgPool) {
         )
       `
     await pgPool.query(insertQuery, [
-      transformedDeals.map(deal => deal.activated_at_epoch),
-      transformedDeals.map(deal => deal.miner_id),
-      transformedDeals.map(deal => deal.client_id),
-      transformedDeals.map(deal => deal.piece_cid),
-      transformedDeals.map(deal => deal.piece_size),
-      transformedDeals.map(deal => deal.term_start_epoch),
-      transformedDeals.map(deal => deal.term_min),
-      transformedDeals.map(deal => deal.term_max),
-      transformedDeals.map(deal => deal.sector_id)
+      activeDeals.map(deal => deal.activated_at_epoch),
+      activeDeals.map(deal => deal.miner_id),
+      activeDeals.map(deal => deal.client_id),
+      activeDeals.map(deal => deal.piece_cid),
+      activeDeals.map(deal => deal.piece_size),
+      activeDeals.map(deal => deal.term_start_epoch),
+      activeDeals.map(deal => deal.term_min),
+      activeDeals.map(deal => deal.term_max),
+      activeDeals.map(deal => deal.sector_id)
     ])
-
-    // Commit the transaction if all inserts are successful
-    console.log(`Inserting ${activeDeals.length} deals took ${Date.now() - startInserting}ms`)
   } catch (error) {
     // If any error occurs, roll back the transaction
     // TODO: Add sentry entry for this error
     // https://github.com/filecoin-station/deal-observer/issues/28
-    console.error('Error inserting deals:', error.message)
+    throw Error('Error inserting deals', { cause: error })
   }
 }
 
 /**
- * @param {Queryable} pgPool
- * @param {string} query
- * @returns {Promise<Array<Static<typeof ActiveDealDbEntry>>>}
- */
-async function loadDeals (pgPool, query) {
+   * @param {Queryable} pgPool
+   * @param {string} query
+   * @returns {Promise<Array<Static <typeof ActiveDealDbEntry>>>}
+   */
+export async function parseDeals (pgPool, query) {
```

Comment:
I have mixed feelings about this functions, it's called `parseDeals` but it queries and parses deals. Maybe it should just parse already queried deals?

Reply:
I believe we agreed to call it `loadDeals`, I am not sure why it's being renamed back to `parseDeals`.